### ISSUE TITLE
Samsung UN55ES6100 TV disconnects while streaming

### ISSUE THREAD

**Issue by Nadahar** [UniversalMediaServer/UniversalMediaServer#754]
There's a [thread in the support forum](http://www.universalmediaserver.com/forum/viewtopic.php?f=9&t=5747&p=21645#p21645) about problems with disconnects after about 15 minutes when streaming to a Samsung TV on wired network. I wouldn't think there was much that could go wrong when streaming over wired network. UMS stops serving because the renderer disconnects, and that in itself looks normal.

The only thing I can find in [the log](http://www.universalmediaserver.com/forum/download/file.php?id=2869) that could possible be an issue is that UMS seems to switch renderer profile during playback. Could this be a problem, that we start out with a working profile and then switches to a profile that doesn't behave with the renderer in question? I'd like for you @skeptical to take a look and see if you understand what happens. Is the `UpnpDetailsSearch = [CD]S?\d{3}\d? , (Samsung|AllShare)` in the `Samsung C/D Series` profile simply casting to wide a net? It's matching `[TV]UN55ES6100 192.168.0.12 uuid:0d1cef00-00dc-1000-9bfe-1c5a3ed197c3 Samsung Electronics UN55ES6100 AllShare1.0 Samsung TV DMR http://www.samsung.com/sec http://www.samsung.com/sec`, but the `(Samsung|AllShare)` will match any device with `Samsung` **or** `AllShare` in it's uPnP details, won't it? That must be a mistake..?

**Comment by skeptical** [comment]
@Nadahar the problem is case-insensitive match in ```RendererConfiguration.matchUPNPDetails()```, where the ```Samsung C/D``` regex:
```ini
UpnpDetailsSearch = [CD]S?\d{3}\d? , (Samsung|AllShare)
```
(after expanding '``` , ```' to '```.*```') gets a hit:
> [TV]UN55ES6100 192.168.0.12 uuid:0d1cef00-00dc-1000-9bfe-1c5a3e**d197c3 Samsung** Electronics UN55ES6100...

I think case-insensitive is obviously counter-productive here, will fix it in a bit when I get a chance (unless you do it first :-).

**Comment by Nadahar** [comment]
@skeptical I can fix it if I am sure about the problem. I though ```,``` was a separator, that each side of the ```,``` became separate search phrases (that comma was in fact **OR**). If it's substituted with ```.*``` it changes things, I think I'd better read the exact code (because I'm now wondering if the spaces are stripped or not). But, do you mean that the uPnP search in general is case insensitive? In that case, isn't it a huge risk to change that? I mean, what about all the current recognitions that will match today that potentially relies on case insenitivity?

Anyway, I'll take a look. But, do you think the renderer profile switch can be the cause of the disconnect? The disconnect happens 5 minutes after the switch in this particular log file.

**Comment by skeptical** [comment]
@Nadahar 

* '``` , ```' is psuedo-code for ```.*```, and was my idea of a readable user-friendly separator so the average regex-unaware user could simply join fragments copied from the ui renderer panel details with it. 
* the case-insensitive part may have been because I re-used the ua matcher code when I wrote it (if I remember right).
* Easiest test would be to remove the regex mismatch and see if the disconnect persists.

**Comment by Nadahar** [comment]
@skeptical 
```,``` makes some sense provided there is a way for users to know this. I see that it's documented in ```DefaultRenderer.conf```:
```
# A good starting point is to use unique parts of '<manufacturer> , <modelName>'
# with each part separated by ' , '. For instance here we use
```
  ...but I'm not sure how many actual users that read it. The question is if it would work just as well to write:
```
# A good starting point is to use unique parts of '<manufacturer> , <modelName>'
# with each part separated by '.*'. For instance here we use
```

Even though case insensitivity wasn't intended, things could rely on it now. But, I'm guessing most uPnP details are copy-pasted and are in the correct case, so we could try to change it and see how much breaks...

I've asked jamesws70 to try to disable the "Samsung C/D Series" renderer profile. Wouldn't that suffice for diagnosis?

**Comment by skeptical** [comment]
@Nadahar yeah, this may be a case of too much spoonfeeding, '``` , ```' should probably be un-(or de-)documented so it can fade into oblivion.  You're right, just instructing to use ```.*``` is better since this also gives the motivated user a starting point for learning what's actually going on.

I doubt if we rely on case-insensitivity at all, since the detail strings are always copied/edited from the logs (or else they wouldn't work, since the order of the fragments matter and the ordering is done by ums, not the device).

**Comment by Nadahar** [comment]
@skeptical I'll throw together a suggestion (already created branch ```RendererSearch```.

**Comment by skeptical** [comment]
@Nadahar no need for a branch imo, these are fairly obvious/simple fixes, but as you prefer of course :-).

**Comment by Nadahar** [comment]
@skeptical I like to do PR's for anything remotely "dangerous" ;)

**Comment by SubJunk** [comment]
Cool to have identified this :)

**Comment by Nadahar** [comment]
@SubJunk Sadly, this doesn't seem to be the (whole) problem. Even though it definitly misidentified the renderer, disabling the "wrong" renderer profile didn't fix the disconnects. Look at the [thread in the support forum] (http://www.universalmediaserver.com/forum/viewtopic.php?f=9&t=5747&p=21645#p21645).

**Comment by Oxalin** [comment]
If I can help at some point at testing something, let me know: I have a Samsung BD-D5100 which disconnects here and there for no obvious explanation (which would set aside the problem about identifying the renderer). It can happen at any moment, sometime it will play fine, but most of the time it will disconnect after a few minutes. Meanwhile, I'll try to compare what is going on between my other devices (that play fine) and that Samsung unit.

**Comment by SubJunk** [comment]
@Oxalin is this still happening with 5.5.0?

**Comment by Oxalin** [comment]
@SubJunk I haven't had time to test it yet. I'll try later today.

**Comment by Optimusprimeums** [comment]
@SubJunk Let me know if i can help, as i own 2 Samsung TV's (UN46ES7100 and UA55D7000)

**Comment by SubJunk** [comment]
@Optimusprimeums have you noticed the issue with your TVs?

**Comment by Oxalin** [comment]
Still having issues: just browsing folders may kill the connection. I'll try to investigate another possible cause first, just in case. I'll test by pluging my computer using RJ45 on the same router as the Samsumg unit. I could also test between Linux and Windows to see if I end up with the same problem. I've been having this stability issue with all the media server softwares I've tried until now, both under Linux and Windows (even Samsung's media server failed on me).

Thus, it may be a different issue than the current one. I'll open a new issue if you're ok with it so we don't pollute the current one any further.

**Comment by patrickbonner** [comment]
Same issue in Samsungs F Series (tried with F9090)

**Comment by Optimusprimeums** [comment]
I'm not seeing these issues but if you have some trace logs so I can compare my Two Samsungs with yours that would be great. Thats @Oxalin and @patrickbonner and sorry if you've already got them in the forums. A link to your post would be great

**Comment by Nadahar** [comment]
Exact cause never found, possibly no longer relevant.
